### PR TITLE
sanitycheck: tool to update testsuite

### PR DESCRIPTION
**Pull request by maksimmasalski**
New tool is created to use with Testrail.
Purpose of that script is to update a test suite in the Testrail,
by uploading test cases metedata using the source .csv file.
(source file is generating by --export-tests sanitycheck option)
1. That tool detects if new sections was added in the source code,
and make them aligned with the test suite in the Testrail system.
2. Script detects if new test cases were added in the source code,
and uploads new test cases to the Testrail.
After run of that tool test suite of the source code
and test suite of the Testrail will be aligned with each other.
Then user can run script report.py to upload results to the Testrail.

Signed-off-by: Maksim Masalski <maksim.masalski@intel.com>

### CODE REVIEW

File: scripts/test_reports/update_testsuite.py [hunk 1]
```diff
@@ -0,0 +1,289 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2019 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# Script to upload results to testrail.
```
Comment (nashif) [inline scripts/test_reports/update_testsuite.py]: this does not match what the script is supposed to be doing.

File: scripts/test_reports/README.md [hunk 2]
```diff
@@ -70,3 +70,41 @@ $ python3 report.py --runner sanitycheck -f <junit file> -V v1.13.0-1386-g183e74
...
+
+# Update Test suite, upload test cases
+
+To upload test results to the TestRail necessary to update Test suite of the TestRail and only then upload test results.
+Update testsuite means to align tests in your Zephyr project with tests in the TestRail, if necessary new test cases and new folders will be added using that script.
+
+1. To receive all test cases from your project run sanitycheck command:
+
+```
+$ sanitycheck --list-tests >> testcases.csv
```
Comment (nashif) [inline scripts/test_reports/README.md]: This does not generate a csv file, use `export-tests`

File: scripts/test_reports/update_testsuite.py [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+from testrail_client import TestRailClient
+
+from junitparser import TestCase, TestSuite, JUnitXml, Skipped, Error
+import re
+import datetime
+import argparse
+import configparser
+import csv
+
+retest_text = "description: subtestcase didn't run: likely the image failed to build or to deploy"
```
Comment (nashif) [inline scripts/test_reports/update_testsuite.py]: how is this relevant?

**Review by nashif — CHANGES_REQUESTED** [review]
Please run the script through pylint and work on improving the score, right now it is 

`Your code has been rated at 3.67/10`

the main function needs to be simplified, too many levels and conditionals.

**Comment by nashif** [comment]
```
i9:test_reports(update_testsuite): pylint  update_testsuite.py
************* Module update_testsuite
update_testsuite.py:61:49: C0326: No space allowed before comma
        self.client.case.add(sec_id, testcase, 4 , 7, 2, None, None, testcase)
                                                 ^ (bad-whitespace)
update_testsuite.py:71:51: C0326: No space allowed before comma
    def find_subsection(self, sections, section_id , subsection):
                                                   ^ (bad-whitespace)
update_testsuite.py:90:57: C0326: Exactly one space required around comparison
            if (s['name'] == section) and (s['parent_id']==None):
                                                         ^^ (bad-whitespace)
update_testsuite.py:111:52: C0326: Exactly one space required around comparison
            if (ns['name'] == section) and (ns['id']==section_id):
                                                    ^^ (bad-whitespace)
update_testsuite.py:116:0: C0301: Line too long (153/100) (line-too-long)
update_testsuite.py:124:0: C0301: Line too long (130/100) (line-too-long)
update_testsuite.py:135:0: C0325: Unnecessary parens after 'if' keyword (superfluous-parens)
update_testsuite.py:136:0: C0325: Unnecessary parens after 'if' keyword (superfluous-parens)
update_testsuite.py:150:0: C0330: Wrong continued indentation (add 15 spaces).
                                        """

For input necessary to use .csv file with the all testcases
generated by the sanitycheck.
                                        """)
                                        ^              | (bad-continuation)
update_testsuite.py:157:0: C0330: Wrong continued indentation (add 17 spaces).
            help="File with the list of all testcases.")
            ^                | (bad-continuation)
update_testsuite.py:169:0: C0325: Unnecessary parens after 'if' keyword (superfluous-parens)
update_testsuite.py:171:0: C0325: Unnecessary parens after 'if' keyword (superfluous-parens)
update_testsuite.py:173:0: C0325: Unnecessary parens after 'if' keyword (superfluous-parens)
update_testsuite.py:178:0: W0311: Bad indentation. Found 12 spaces, expected 8 (bad-indentation)
update_testsuite.py:179:0: W0311: Bad indentation. Found 20 spaces, expected 12 (bad-indentation)
update_testsuite.py:196:13: C0326: Exactly one space required around assignment
        suite=run.receive_suite()
             ^ (bad-whitespace)
update_testsuite.py:197:16: C0326: Exactly one space required around assignment
        sections=run.receive_sections()
                ^ (bad-whitespace)
update_testsuite.py:202:14: C0326: Exactly one space required before assignment
        reader= csv.reader(csvfile, delimiter=',')
              ^ (bad-whitespace)
update_testsuite.py:217:0: C0301: Line too long (114/100) (line-too-long)
update_testsuite.py:229:0: C0301: Line too long (102/100) (line-too-long)
update_testsuite.py:230:0: C0301: Line too long (107/100) (line-too-long)
update_testsuite.py:238:0: C0301: Line too long (118/100) (line-too-long)
update_testsuite.py:238:84: C0326: No space allowed before comma
                                    run.client.case.add(subsec_id, testcase_name, 4 , 7, 2, None, None, testcase_name)
                                                                                    ^ (bad-whitespace)
update_testsuite.py:241:0: C0301: Line too long (133/100) (line-too-long)
update_testsuite.py:242:0: C0301: Line too long (110/100) (line-too-long)
update_testsuite.py:243:0: C0301: Line too long (111/100) (line-too-long)
update_testsuite.py:246:0: C0301: Line too long (118/100) (line-too-long)
update_testsuite.py:246:84: C0326: No space allowed before comma
                                    run.client.case.add(subsec_id, testcase_name, 4 , 7, 2, None, None, testcase_name)
                                                                                    ^ (bad-whitespace)
update_testsuite.py:250:0: C0301: Line too long (126/100) (line-too-long)
update_testsuite.py:252:0: C0301: Line too long (107/100) (line-too-long)
update_testsuite.py:254:0: C0301: Line too long (106/100) (line-too-long)
update_testsuite.py:255:0: C0301: Line too long (107/100) (line-too-long)
update_testsuite.py:258:0: C0301: Line too long (114/100) (line-too-long)
update_testsuite.py:258:80: C0326: No space allowed before comma
                                run.client.case.add(subsec_id, testcase_name, 4 , 7, 2, None, None, testcase_name)
                                                                                ^ (bad-whitespace)
update_testsuite.py:261:0: C0301: Line too long (119/100) (line-too-long)
update_testsuite.py:263:0: C0301: Line too long (116/100) (line-too-long)
update_testsuite.py:266:0: C0301: Line too long (111/100) (line-too-long)
update_testsuite.py:266:77: C0326: No space allowed before comma
                                run.client.case.add(sec_id, testcase_name, 4 , 7, 2, None, None, testcase_name)
                                                                             ^ (bad-whitespace)
update_testsuite.py:269:0: C0301: Line too long (111/100) (line-too-long)
update_testsuite.py:271:0: C0301: Line too long (107/100) (line-too-long)
update_testsuite.py:274:0: C0301: Line too long (111/100) (line-too-long)
update_testsuite.py:274:77: C0326: No space allowed before comma
                                run.client.case.add(sec_id, testcase_name, 4 , 7, 2, None, None, testcase_name)
                                                                             ^ (bad-whitespace)
update_testsuite.py:278:0: W0301: Unnecessary semicolon (unnecessary-semicolon)
update_testsuite.py:1:0: C0111: Missing module docstring (missing-docstring)
update_testsuite.py:15:0: C0413: Import "from testrail_client import TestRailClient" should be placed at the top of the module (wrong-import-position)
update_testsuite.py:17:0: C0413: Import "from junitparser import TestCase, TestSuite, JUnitXml, Skipped, Error" should be placed at the top of the module (wrong-import-position)
update_testsuite.py:18:0: C0413: Import "import re" should be placed at the top of the module (wrong-import-position)
update_testsuite.py:19:0: C0413: Import "import datetime" should be placed at the top of the module (wrong-import-position)
update_testsuite.py:20:0: C0413: Import "import argparse" should be placed at the top of the module (wrong-import-position)
update_testsuite.py:21:0: C0413: Import "import configparser" should be placed at the top of the module (wrong-import-position)
update_testsuite.py:22:0: C0413: Import "import csv" should be placed at the top of the module (wrong-import-position)
update_testsuite.py:24:0: C0103: Constant name "retest_text" doesn't conform to UPPER_CASE naming style (invalid-name)
update_testsuite.py:28:0: C0103: Constant name "err_messages" doesn't conform to UPPER_CASE naming style (invalid-name)
update_testsuite.py:29:0: C0103: Constant name "testcases_counter" doesn't conform to UPPER_CASE naming style (invalid-name)
update_testsuite.py:30:0: C0103: Constant name "testcases_testrail" doesn't conform to UPPER_CASE naming style (invalid-name)
update_testsuite.py:31:0: C0103: Constant name "testcases_error" doesn't conform to UPPER_CASE naming style (invalid-name)
update_testsuite.py:33:0: C0111: Missing function docstring (missing-docstring)
update_testsuite.py:37:0: C0111: Missing class docstring (missing-docstring)
update_testsuite.py:45:4: C0111: Missing method docstring (missing-docstring)
update_testsuite.py:46:8: W0201: Attribute 'project' defined outside __init__ (attribute-defined-outside-init)
update_testsuite.py:47:8: W0201: Attribute 'user' defined outside __init__ (attribute-defined-outside-init)
update_testsuite.py:48:8: W0201: Attribute 'token' defined outside __init__ (attribute-defined-outside-init)
update_testsuite.py:37:0: R0903: Too few public methods (1/2) (too-few-public-methods)
update_testsuite.py:52:0: C0111: Missing class docstring (missing-docstring)
update_testsuite.py:54:4: W0231: __init__ method from base class 'TestRail' is not called (super-init-not-called)
update_testsuite.py:60:4: C0111: Missing method docstring (missing-docstring)
update_testsuite.py:63:4: C0111: Missing method docstring (missing-docstring)
update_testsuite.py:67:4: C0111: Missing method docstring (missing-docstring)
update_testsuite.py:71:4: C0111: Missing method docstring (missing-docstring)
update_testsuite.py:74:12: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)
update_testsuite.py:78:8: R1705: Unnecessary "else" after "return" (no-else-return)
update_testsuite.py:71:4: R0201: Method could be a function (no-self-use)
update_testsuite.py:86:4: C0111: Missing method docstring (missing-docstring)
update_testsuite.py:89:12: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)
update_testsuite.py:90:43: C0121: Comparison to None should be 'expr is None' (singleton-comparison)
update_testsuite.py:93:8: R1705: Unnecessary "else" after "return" (no-else-return)
update_testsuite.py:86:4: R0201: Method could be a function (no-self-use)
update_testsuite.py:102:4: C0111: Missing method docstring (missing-docstring)
update_testsuite.py:103:12: C0103: Variable name "c" doesn't conform to snake_case naming style (invalid-name)
update_testsuite.py:102:4: R0201: Method could be a function (no-self-use)
update_testsuite.py:108:4: C0111: Missing method docstring (missing-docstring)
update_testsuite.py:110:12: C0103: Variable name "ns" doesn't conform to snake_case naming style (invalid-name)
update_testsuite.py:114:8: R1705: Unnecessary "else" after "return" (no-else-return)
update_testsuite.py:123:4: C0111: Missing method docstring (missing-docstring)
update_testsuite.py:125:8: R1705: Unnecessary "else" after "return" (no-else-return)
update_testsuite.py:123:26: W0613: Unused argument 'sections' (unused-argument)
update_testsuite.py:132:0: C0111: Missing function docstring (missing-docstring)
update_testsuite.py:135:7: C0121: Comparison to None should be 'expr is None' (singleton-comparison)
update_testsuite.py:136:8: R1705: Unnecessary "else" after "return" (no-else-return)
update_testsuite.py:136:11: C0121: Comparison to None should be 'expr is None' (singleton-comparison)
update_testsuite.py:145:0: C0111: Missing function docstring (missing-docstring)
update_testsuite.py:166:0: C0111: Missing function docstring (missing-docstring)
update_testsuite.py:177:7: C1801: Do not use `len(SEQUENCE)` to determine if a sequence is empty (len-as-condition)
update_testsuite.py:178:12: C0200: Consider using enumerate instead of iterating with range and len (consider-using-enumerate)
update_testsuite.py:185:0: C0111: Missing function docstring (missing-docstring)
update_testsuite.py:185:0: R0914: Too many local variables (18/15) (too-many-locals)
update_testsuite.py:198:8: C0103: Constant name "testcases_counter" doesn't conform to UPPER_CASE naming style (invalid-name)
update_testsuite.py:198:8: W0603: Using the global statement (global-statement)
update_testsuite.py:199:8: C0103: Constant name "testcases_testrail" doesn't conform to UPPER_CASE naming style (invalid-name)
update_testsuite.py:199:8: W0603: Using the global statement (global-statement)
update_testsuite.py:200:8: C0103: Constant name "testcases_error" doesn't conform to UPPER_CASE naming style (invalid-name)
update_testsuite.py:200:8: W0603: Using the global statement (global-statement)
update_testsuite.py:279:12: W0702: No exception type(s) specified (bare-except)
update_testsuite.py:203:8: R1702: Too many nested blocks (7/5) (too-many-nested-blocks)
update_testsuite.py:283:16: C0103: Constant name "files_error" doesn't conform to UPPER_CASE naming style (invalid-name)
update_testsuite.py:283:16: W0602: Using global for 'files_error' but no assignment is done (global-variable-not-assigned)
update_testsuite.py:203:8: R1702: Too many nested blocks (6/5) (too-many-nested-blocks)
update_testsuite.py:242:36: W0612: Unused variable 'add_subsection' (unused-variable)
update_testsuite.py:251:32: W0612: Unused variable 'add_sec' (unused-variable)
update_testsuite.py:185:0: R0912: Too many branches (13/12) (too-many-branches)
update_testsuite.py:185:0: R0915: Too many statements (78/50) (too-many-statements)
update_testsuite.py:12:0: W0611: Unused import glob (unused-import)
update_testsuite.py:13:0: W0611: Unused pprint imported from pprint (unused-import)
update_testsuite.py:17:0: W0611: Unused TestCase imported from junitparser (unused-import)
update_testsuite.py:17:0: W0611: Unused TestSuite imported from junitparser (unused-import)
update_testsuite.py:17:0: W0611: Unused JUnitXml imported from junitparser (unused-import)
update_testsuite.py:17:0: W0611: Unused Skipped imported from junitparser (unused-import)
update_testsuite.py:17:0: W0611: Unused Error imported from junitparser (unused-import)
update_testsuite.py:19:0: W0611: Unused import datetime (unused-import)
update_testsuite.py:21:0: W0611: Unused import configparser (unused-import)
update_testsuite.py:17:0: C0411: third party import "from junitparser import TestCase, TestSuite, JUnitXml, Skipped, Error" should be placed before "from testrail_client import TestRailClient" (wrong-import-order)
update_testsuite.py:18:0: C0411: standard import "import re" should be placed before "from junitparser import TestCase, TestSuite, JUnitXml, Skipped, Error" (wrong-import-order)
update_testsuite.py:19:0: C0411: standard import "import datetime" should be placed before "from junitparser import TestCase, TestSuite, JUnitXml, Skipped, Error" (wrong-import-order)
update_testsuite.py:20:0: C0411: standard import "import argparse" should be placed before "from junitparser import TestCase, TestSuite, JUnitXml, Skipped, Error" (wrong-import-order)
update_testsuite.py:21:0: C0411: standard import "import configparser" should be placed before "from junitparser import TestCase, TestSuite, JUnitXml, Skipped, Error" (wrong-import-order)
update_testsuite.py:22:0: C0411: standard import "import csv" should be placed before "from junitparser import TestCase, TestSuite, JUnitXml, Skipped, Error" (wrong-import-order)

-----------------------------------
Your code has been rated at 3.67/10
```

**Comment by xuhao8210** [comment]
_In the file:” update_testsuite.py”, I have few comments below ,could you please review them ? Thanks._

_1. Could you merge 2 functions: find_subsection() & find_section() into one function, and please review the new function following, you may enhance it if need._ 
     =======================================================
```
      def find_section(self, sections, section, section_id=None):
            for sec in sections:
                if (sec['name'] == section) and (sec['parent_id'] == section_id):
	                print("section or Subsection ID obtained: ", sec['id'])
		        return sec['id']

            print("section or Subsection is not obtained. Need to add new one.")
            return None
```
    =======================================================
_2. For the function:” update_testsuite”, add the try and catch code to catch the exception for each row from csvfile (e.g. testcase name is blank , or whole row is blank)_

_3. For the function:” update_testsuite”, update some logical code from “if-else” to “if not”, please review an example following:_

========================old code ===============================
```
              subsec_id = run.find_subsection(sections, sec_id, subsection)
              print("subsec_id: ", subsec_id)
               if subsec_id:`
                       print("=> Adding testcase: ", testcase_name)
                       run.add_testcase(subsec_id, testcase_name)
                      testcases_counter += 1
              else:
                       print(" ==> No subsection found. Creating {}/{}".format(section, subsection))
                       run.add_subsection(sections, section, sec_id, subsection)
                       sections = run.client.section.for_suite(run.project_id, suite_id=run.suite)
                       subsec_id = run.find_subsection(sections, sec_id, subsection)
                       print("=> Adding testcase: ", testcase_name)
                       run.add_testcase(subsec_id, testcase_name)
                       testcases_counter += 1
```

========================update to ===============================
```
            subsec_id = run.find_subsection(sections, sec_id, subsection)
            print("subsec_id: ", subsec_id)		
             if not subsec_id:
                 print(" ==> No subsection found. Creating {}/{}".format(section, subsection))
                 run.add_subsection(sections, section, sec_id, subsection)
                 sections = run.client.section.for_suite(run.project_id,suite_id=run.suite)
	         subsec_id = run.find_subsection(sections, sec_id, subsection)
															
             print("=> Adding testcase: ", testcase_name)`
             run.add_testcase(subsec_id, testcase_name)
             testcases_counter += 1
```
===========================================================

**Comment by maksimmasalski** [comment]
> _In the file:” update_testsuite.py”, I have few comments below ,could you please review them ? Thanks._
> 
> _1. Could you merge 2 functions: find_subsection() & find_section() into one function, and please review the new function following, you may enhance it if need._
> =======================================================
> 
> ```
>       def find_section(self, sections, section, section_id=None):
>             for sec in sections:
>                 if (sec['name'] == section) and (sec['parent_id'] == section_id):
> 	                print("section or Subsection ID obtained: ", sec['id'])
> 		        return sec['id']
> 
>             print("section or Subsection is not obtained. Need to add new one.")
>             return None
> ```
> 
> ```
> =======================================================
> ```
> 
> _2. For the function:” update_testsuite”, add the try and catch code to catch the exception for each row from csvfile (e.g. testcase name is blank , or whole row is blank)_
> 
> _3. For the function:” update_testsuite”, update some logical code from “if-else” to “if not”, please review an example following:_
> 
> ========================old code ===============================
> 
> ```
>               subsec_id = run.find_subsection(sections, sec_id, subsection)
>               print("subsec_id: ", subsec_id)
>                if subsec_id:`
>                        print("=> Adding testcase: ", testcase_name)
>                        run.add_testcase(subsec_id, testcase_name)
>                       testcases_counter += 1
>               else:
>                        print(" ==> No subsection found. Creating {}/{}".format(section, subsection))
>                        run.add_subsection(sections, section, sec_id, subsection)
>                        sections = run.client.section.for_suite(run.project_id, suite_id=run.suite)
>                        subsec_id = run.find_subsection(sections, sec_id, subsection)
>                        print("=> Adding testcase: ", testcase_name)
>                        run.add_testcase(subsec_id, testcase_name)
>                        testcases_counter += 1
> ```
> 
> ========================update to ===============================
> 
> ```
>             subsec_id = run.find_subsection(sections, sec_id, subsection)
>             print("subsec_id: ", subsec_id)		
>              if not subsec_id:
>                  print(" ==> No subsection found. Creating {}/{}".format(section, subsection))
>                  run.add_subsection(sections, section, sec_id, subsection)
>                  sections = run.client.section.for_suite(run.project_id,suite_id=run.suite)
> 	         subsec_id = run.find_subsection(sections, sec_id, subsection)
> 															
>              print("=> Adding testcase: ", testcase_name)`
>              run.add_testcase(subsec_id, testcase_name)
>              testcases_counter += 1
> ```
> 
> ===========================================================

@xuhao8210 Please review my code again. For the (2.) what you suggested, I added into Error 1 number of the raw. We have one exception in main function for function update_testsuite, don't think that necessary to add more exceptions.

**Comment by xuhao8210** [comment]
> > _In the file:” update_testsuite.py”, I have few comments below ,could you please review them ? Thanks._
> > # _1. Could you merge 2 functions: find_subsection() & find_section() into one function, and please review the new function following, you may enhance it if need._
> > ```
> >       def find_section(self, sections, section, section_id=None):
> >             for sec in sections:
> >                 if (sec['name'] == section) and (sec['parent_id'] == section_id):
> > 	                print("section or Subsection ID obtained: ", sec['id'])
> > 		        return sec['id']
> > 
> >             print("section or Subsection is not obtained. Need to add new one.")
> >             return None
> > ```
> > 
> > 
> > ```
> > =======================================================
> > ```
> > 
> > 
> > _2. For the function:” update_testsuite”, add the try and catch code to catch the exception for each row from csvfile (e.g. testcase name is blank , or whole row is blank)_
> > _3. For the function:” update_testsuite”, update some logical code from “if-else” to “if not”, please review an example following:_
> > ========================old code ===============================
> > ```
> >               subsec_id = run.find_subsection(sections, sec_id, subsection)
> >               print("subsec_id: ", subsec_id)
> >                if subsec_id:`
> >                        print("=> Adding testcase: ", testcase_name)
> >                        run.add_testcase(subsec_id, testcase_name)
> >                       testcases_counter += 1
> >               else:
> >                        print(" ==> No subsection found. Creating {}/{}".format(section, subsection))
> >                        run.add_subsection(sections, section, sec_id, subsection)
> >                        sections = run.client.section.for_suite(run.project_id, suite_id=run.suite)
> >                        subsec_id = run.find_subsection(sections, sec_id, subsection)
> >                        print("=> Adding testcase: ", testcase_name)
> >                        run.add_testcase(subsec_id, testcase_name)
> >                        testcases_counter += 1
> > ```
> > 
> > 
> > ========================update to ===============================
> > ```
> >             subsec_id = run.find_subsection(sections, sec_id, subsection)
> >             print("subsec_id: ", subsec_id)		
> >              if not subsec_id:
> >                  print(" ==> No subsection found. Creating {}/{}".format(section, subsection))
> >                  run.add_subsection(sections, section, sec_id, subsection)
> >                  sections = run.client.section.for_suite(run.project_id,suite_id=run.suite)
> > 	         subsec_id = run.find_subsection(sections, sec_id, subsection)
> > 															
> >              print("=> Adding testcase: ", testcase_name)`
> >              run.add_testcase(subsec_id, testcase_name)
> >              testcases_counter += 1
> > ```
> > 
> > 
> > ===========================================================
> 
> @xuhao8210 Please review my code again. For the (2.) what you suggested, I added into Error 1 number of the raw. We have one exception in main function for function update_testsuite, don't think that necessary to add more exceptions.

@maksimmasalski  That is fine after I reviewed them again. Thanks.

File: scripts/test_reports/update_testsuite.py [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+                run.add_testcase(sec_id, testcase_name)
+                testcases_counter += 1
+
+            print("=> Adding testcase: ", testcase_name)
+            run.add_testcase(sec_id, testcase_name)
+            testcases_counter += 1
+
+def main():
+    args = parse_args()
+    run = TestSuiteUpdate(results_file=args.results_file, project_id=args.project, suite=args.suite)
```
Comment (nashif) [inline scripts/test_reports/update_testsuite.py:234]: You need to check for the arguments before you use them, right now if I run the script without any arguments I get:

```
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): zephyrproject.testrail.io:443
DEBUG:urllib3.connectionpool:https://zephyrproject.testrail.io:443 "GET /index.php?/api/v2/get_cases/None&suite_id=None&sectio8
Traceback (most recent call last):
  File "scripts/test_reports/update_testsuite.py", line 271, in <module>
    main()
  File "scripts/test_reports/update_testsuite.py", line 242, in main
    cases_suite = run.receive_cases_suite()
  File "scripts/test_reports/update_testsuite.py", line 46, in receive_cases_suite
    cases_suite = self.client.case.for_project(project_id=self.project_id, suite_id=self.suite)
  File "/Users/anashif/zephyr/zephyrproject/qm/scripts/test_reports/testrail_client/api/case.py", line 37, in for_project
    params=filters)
  File "/Users/anashif/zephyr/zephyrproject/qm/scripts/test_reports/testrail_client/api/base.py", line 49, in _format
    return func(*args, **kwargs)
  File "/Users/anashif/zephyr/zephyrproject/qm/scripts/test_reports/testrail_client/api/base.py", line 35, in _check
    raise TesRailAPIError(resp)
testrail_client.api.error.TesRailAPIError: [TestRailAPI] 400 || (Bad Request) || b'{"error":"Field :project_id is not a valid '
```

File: scripts/test_reports/update_testsuite.py [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+                                                       """
+
+    For input necessary to use .csv file with the all testcases
+    generated by the sanitycheck.
+                                        """)
+    xor_results = input_testcases_select.add_mutually_exclusive_group()
+    xor_results.add_argument('-f', '--results-file', default=None,
+                             help="File with the list of all testcases.")
+
+    parser.add_argument('-p', '--project', type=int, help="Project ID")
```
Comment (nashif) [inline scripts/test_reports/update_testsuite.py]: this should be a require option, no?

File: scripts/test_reports/update_testsuite.py [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+    parser.add_argument('-s', '--suite', type=int, help="Suite ID")
```
Comment (nashif) [inline scripts/test_reports/update_testsuite.py]: same here

**Review by nashif — CHANGES_REQUESTED** [review]
- The script is new, no need to submit this in multiple commits. The fixups and changes based on pylint should not come as a followup commit, instead they need to be in one commit. (You should not introduce an issue in 1 commit and fix it in the next one, this makes review very difficult).
- the commit messages are not accurate, still talking about the old approach of uploading results and creating sections for missing tests
- I tried the script on an empty testsuite, there is a major problem with the script, see https://zephyrproject.testrail.io/index.php?/suites/view/61&group_by=cases:section_id&group_order=asc&group_id=5357, You are creating a lot of duplicate sections.

**Comment by maksimmasalski** [comment]
>     * The script is new, no need to submit this in multiple commits. The fixups and changes based on pylint should not come as a followup commit, instead they need to be in one commit. (You should not introduce an issue in 1 commit and fix it in the next one, this makes review very difficult).
> 
>     * the commit messages are not accurate, still talking about the old approach of uploading results and creating sections for missing tests
> 
>     * I tried the script on an empty testsuite, there is a major problem with the script, see https://zephyrproject.testrail.io/index.php?/suites/view/61&group_by=cases:section_id&group_order=asc&group_id=5357, You are creating a lot of duplicate sections.

@nashif yes, I made mistake, when I merged two functions together, I forgot to add line to update sections. It cause problem when new section added, next run always created duplicate of that section, because variable `sections` had obsolete list of the sections. Fixed it. The same problem was for cases, it created duplicate of test cases too.
Also, I set argparser to ask for the necessary options.

**Review by nashif — APPROVED** [review]

**Pull request merged**